lib/board.rb

```diff
@@ -28,61 +28,19 @@ def invert_board
     end
   end
   
-  def invert_board_diagonal
-    n = 0
-    i = 0
-    7.times do
-      diagonal = []
-      @horizontal_board.values.reverse.each do |row|
-        diagonal << row[i]
-        i += 1
-      end
-      @diagonal_board[n] = diagonal
-      n += 1
-      i = n
-    end
-    
-    k = 0
-    i = 0
-    7.times do
-      diagonal = []
-      @horizontal_board.values.each do |row|
-        diagonal << row[i]
-        i += 1
-      end
-      @diagonal_board[n] = diagonal
-      n += 1
-      k += 1
-      i = k
-    end
-    
-    k = 4
-    i = 4
-    5.times do
-      diagonal = []
-      @horizontal_board.values.reverse.each do |row|
-        diagonal << row[i] if i >= 0
-        i -= 1
-      end
-      @diagonal_board[n] = diagonal
-      n += 1
-      k -= 1
-      i = k
-    end
-    
-    k = 4
-    i = 4
-    5.times do
-      diagonal = []
-      @horizontal_board.values.each do |row|
-        diagonal << row[i] if i >= 0
-        i -= 1
-      end
-      @diagonal_board[n] = diagonal
-      n += 1
-      k -= 1
-      i = k
-    end
+  def invert_board_diagonal(grid)
```

Comment:
VERY COOL how you were able to condense the method! I'd like for you to walk me through it during the week. 
also, would replacing the letters with "index" or "column" (b/c i assume that's what "j" stands for?) make it more readable?